frontend/src/routes/Applications/Overview.tsx

```diff
@@ -349,10 +349,9 @@ export default function ApplicationsOverview() {
                 .filter((argoApp) => {
                     const resources = argoApp.status ? argoApp.status.resources : undefined
                     let definedNamespace = ''
-                    resources &&
-                        resources.forEach((resource: any) => {
-                            definedNamespace = resource.namespace
-                        })
+                    if (resources) {
```

Comment:
`resources` has type `any`, so we shouldn't really make any assumptions about its type. However, it appears to be understood that it is an array. In the old code, we assumed it was an array. In the new code, we assume it is an array AND that it has elements. 

Try this in the console:
`if ([]) console.log("truthy")`

We should at least check that `resources` has 1 element before we try to access it.

Reply:
Sounds good, thank @KevinFCormier , will make the change :)

Reply:
Also, could these resources be cluster-scoped and not have a namespace? The previous code would not have handled that either, but it would have taken the value from the last resource instead of the first.

Reply:
Yeah they could be. For now the namespaces do exist, but I guess they could be possible to not have namespace. Given that, maybe for each was better?

Reply:
Oh I can use get, and set it to undefined by default.

If no definedNamespace, it will use other spec

```                    definedNamespace ? definedNamespace : argoApp.spec.destination.namespace```